pkg/controller/managedresource/add.go

```diff
@@ -55,12 +58,24 @@ type ControllerConfig struct {
 	SyncPeriod           time.Duration
 	ClassFilter          *filter.ClassFilter
 	AlwaysUpdate         bool
+	ClusterID            string
 
 	TargetClientConfig resourcemanagercmd.TargetClientConfig
 }
 
 // AddToManagerWithOptions adds the controller to a Manager with the given config.
 func AddToManagerWithOptions(mgr manager.Manager, conf ControllerConfig) error {
+	if conf.ClusterID == "<cluster>" || conf.ClusterID == "<default>" {
+		mgr.GetLogger().Info("Trying to get cluster id from cluster")
+		tmpClient, err := client.New(mgr.GetConfig(), client.Options{})
+		if err == nil {
+			conf.ClusterID, err = determineClusterIdentity(tmpClient, conf.ClusterID == "<cluster>")
+		}
+		if err != nil {
+			return fmt.Errorf("unable to determine cluster id: %+v", err)
+		}
+	}
+	mgr.GetLogger().Info("Used cluster id: " + conf.ClusterID)
```

Comment:
I guess this is better to be moved [here](https://github.com/gardener/gardener-resource-manager/blob/master/cmd/gardener-resource-manager/app/app.go#L78).

Then you can implement a function like https://github.com/gardener/gardener-resource-manager/blob/4c0643b911024fb0dfb2d26a051257f4a1f5aeb4/pkg/controller/managedresource/add.go#L137-L139

and call it. This way you can also re-use the context from the cobra cmd and do not have to use `context.Background()`.

Reply:
Hmm, is a little more complicate. I've done it, but I'm not sure whether I'm convinced,